test/ao/sparsity/test_structured_sparsifier.py

```diff
@@ -75,6 +95,72 @@ def test_update_mask(self):
         assert expected.shape == pruned.shape
         assert torch.isclose(expected, pruned, rtol=1e-05, atol=1e-07).all()
 
+    def test_lstm_saliency_pruner_update_mask(self):
```

Comment:
@WongJohnson I added this test for the SaliencyPruner and  two additional tests for LSTM -> layernorm -> Linear below.

Reply:
Great, thanks for adding this. I'd like to see a bit more complex tests (perhaps randomly-drawn tensors) but I'll leave that up to your team's judgement. I would at least ensure that you test properly handling negative weights and also 1d tensors (i.e. bias tensors for which I had to add the dims() check - I believe `SaliencyPruner` still has this issue).

Reply:
I think actually pruning 1d tensors is something that should not work - since we do rowise pruning, it doesn't really make sense to prune a 1d tensor. When we prune the bias, we're actually just taking the mask from the weight and applying it to the bias tensor, not computing a mask for the bias itself. This should probably throw a clearer error message earlier though.